src/LinAlg/hiopLinSolverSparseGinkgo.cpp

```diff
@@ -232,28 +232,46 @@ std::shared_ptr<gko::Executor> create_exec(std::string executor_string)
 }
 
 
+gko::solver::trisolve_algorithm create_alg(std::string algorithm_string)
+{
+    std::map<std::string, gko::solver::trisolve_algorithm>
+        alg_map{
+            {"syncfree", gko::solver::trisolve_algorithm::syncfree},
+            {"sparselib", gko::solver::trisolve_algorithm::sparselib}};
+
+    return alg_map.at(algorithm_string);
+}
+
+
 std::shared_ptr<gko::LinOpFactory> setup_solver_factory(std::shared_ptr<const gko::Executor> exec,
-                                                        std::shared_ptr<gko::matrix::Csr<double, int>> mtx)
+                                                        std::shared_ptr<gko::matrix::Csr<double, int>> mtx,
+                                                        gko::solver::trisolve_algorithm alg,
+                                                        const unsigned gmres_iter, const double gmres_tol, const unsigned gmres_restart)
 {
     auto preprocessing_fact = gko::share(gko::reorder::Mc64<double, int>::build().on(exec));
     auto preprocessing = gko::share(preprocessing_fact->generate(mtx));
     auto lu_fact = gko::share(gko::experimental::factorization::Glu<double, int>::build_reusable()
                               .on(exec, mtx.get(), preprocessing.get()));
     auto inner_solver_fact = gko::share(gko::experimental::solver::Direct<double, int>::build()
                                         .with_factorization(lu_fact)
+                                        .with_algorithm(alg)
                                         .on(exec));
-    auto solver_fact = gko::share(gko::solver::Gmres<>::build()
-                                  .with_criteria(
+
+    std::shared_ptr<gko::LinOpFactory> solver_fact = inner_solver_fact;
+    if (gmres_iter > 0) {
+        solver_fact = gko::share(gko::solver::Gmres<double>::build()
```

Comment:
misaligned arguments

Reply:
looks better; arguments of `gko::share` appears to be misaligned, but no place to get too pedantic.